src/core/Core.js

```diff
@@ -415,61 +422,93 @@ class Uppy {
     })
   }
 
-  pauseResume (action, fileID) {
+  pauseResume (fileID) {
+    const updatedFiles = Object.assign({}, this.getState().files)
+
+    if (updatedFiles[fileID].uploadComplete) return
+
+    const wasPaused = updatedFiles[fileID].isPaused || false
+    const isPaused = !wasPaused
+
+    let updatedFile
+    if (wasPaused) {
+      updatedFile = Object.assign({}, updatedFiles[fileID], {
+        isPaused: false
+      })
+    } else {
+      updatedFile = Object.assign({}, updatedFiles[fileID], {
+        isPaused: true
+      })
+    }
```

Comment:
Not really related, but this `if` could be replaced with

```js
const updatedFile = Object.assign({}, updatedFiles[fileID], {
  isPaused: !wasPaused
})
```

Reply:
It’s very related, it helps me grow as a human. Thanks!